templates/patroni.yml.j2

```diff
@@ -52,6 +52,7 @@ bootstrap:
     command: >
       pgbackrest --stanza={{ restore_stanza }} --pg1-path={{ storage_path }}/pgdata
       {%- if backup_id %} --set={{ backup_id }} {%- endif %}
+      {%- if restore_timeline %} --target-timeline="{{ restore_timeline }}" {% endif %}
```

Comment:
Doesn't `current` work as the value for the `target-timeline` parameter? If so, we could eliminate the need to pass a timeline as the backup ID in the restore action and match what was agreed upon in the specification document.

Reply:
Do you mean the case when user uses real backup-id (with `action: full backup`, for example) and specifies restore-to-time? Or do you mean to eliminate ability to pass `action: restore` to the backup-id at the restore?

Reply:
I mean letting the user be able to pass only the `restore-to-time` parameter (without requiring the user to pass the `backup-id` parameter too). I understand that the available timestamps we can restore to are unique among all the timelines because after we restore either to a backup-id or to a restore-to-time timestamp, the charm will generate a new timeline, so the new data will be written with the current datetime, not some timestamp from the past (that would be within the other timelines). With that, it seems that we'd be able to pass only the `restore-to-time` parameter to the restore action. Does that make sense?

I was wondering whether passing `current ` to the `target-timeline` parameter in the Patroni configuration file template would work to always restore to the right point in time, no matter what timeline it belongs to.

Reply:
`--target-timeline=current` will just set restoring timeline to the one that was active when chosen backup was created. And if there is no backup-id passed (`--set` parameter), PostgreSQL / pgBackRest will choose nearest backup before specified restore-to-time (`--target` parameter). And it will lead to very inconvenient UI as user will not always create backup after each restore. I think such functionality can take place when something like automatic backups creation system is introduced.

For example, in the PITR integration test algorithm only one single full backup used. In such situation user will be able to restore only along timeline 1.

I will attach additional diagram to visualize this issue.

![image](https://github.com/user-attachments/assets/317b8f67-efd0-407a-befc-3c0fe3c6d9c7)

Reply:
Got it. Thanks for checking that. I'll just sync with Mohamed (our PR) about this requirement to see whether we need to change anything else.

Reply:
Copied from Matrix chat:

Do you think that the following approach would allow PITR with only the timestamp parameter (without requiring the user to pass the backup-id):
* Get the PITR timestamp.
* List backups by start time (timestamp) and restores by end time (timestamp).
* Find the nearest backup/restore with the timestamp before the PITR timestamp.
* Get the timeline ID of that backup/restore.
* Convert the timeline ID to hexadecimal (like mentioned at https://postgresqlco.nf/doc/en/param/recovery_target_timeline/).
* Pass the hexadecimal timeline ID to the --target-timeline parameter.

Vladyslav will test that later.

Reply:
Added this functionality in both K8s and VM PRs.